packages/oruga/src/composables/useInputHandler.ts

```diff
@@ -225,6 +226,45 @@ export function useInputHandler<T extends ValidatableFormElement>(
     }
 
     if (!isSSR) {
+        /// Propagates any custom constraint validation message to the underlying DOM element
+        const updateCustomValidationMessage = (): void => {
+            if (!(props.useHtml5Validation ?? true)) {
+                return;
+            }
+            const element = maybeElement.value;
+            if (!isDefined(element)) {
+                return;
+            }
+            const validity = props.customValidity ?? "";
+            if (typeof validity === "string") {
+                element.setCustomValidity(validity);
+            } else {
+                element.setCustomValidity(validity(element.validity));
+            }
+        };
+        // Note that using watchEffect will implicitly pick up any reactive dependencies used
+        // inside props.customValidity, which should help the computed message stay up to date.
+        watchEffect(updateCustomValidationMessage);
+
+        // Clean up validation state if we stop controlling it.
+        watch(
+            [maybeElement, (): boolean => props.useHtml5Validation ?? true],
```

Comment:
Do you have a reason to have two watchers watching the same objects? Or just cleaner code? Maybe performance wise it is better to merge them together?

Reply:
I hadn't noticed that they were watching the same objects, but I think it's cleaner to have separate watchers; they have mostly independent purposes, and mingling their logic would make the second watcher even harder to read. The second watcher also has different timing (`{ immediate: true }`).

Especially with how rarely these watchers are actually triggered, I expect the performance differences to be negligible.